search/includes/classes/class-search.php

```diff
@@ -389,6 +389,13 @@ protected function setup_hooks() {
 
 		// Do not show the above compat notice since VIP Search will support whatever Elasticsearch version we're running
 		add_filter( 'pre_option_ep_hide_es_above_compat_notice', '__return_true' );
+
+		// If protected content is enabled, ensure that the attachment post type is an indexable post type.
+		// Set the priority to PHP_INT_MAX - 1 so customers can use PHP_INT_MAX to unset it if needed.
+		if ( false !== $this->is_protected_content_enabled() ) {
+			$priority = PHP_INT_MAX - 1;
```

Comment:
PHP_INT_MAX (the priority needed to override this) is pretty final - is there a lower priority we can use that leaves more flexibility for customers to override?